apps/api/src/app/events/usecases/send-message/send-message.usecase.ts

```diff
@@ -174,11 +175,15 @@ export class SendMessage {
       environmentId: job._environmentId,
     });
 
+    const subscriber = await this.subscriberRepository.findById(job._subscriberId);
+    if (!subscriber) throw new ApiException('Subscriber not found with id ' + job._subscriberId);
```

Comment:
Btw I am not sure we need to add the exception here because if the flow got till here then we are after ProcessSubscriber usecase, in that case, the execute function already fetched/created a subscriber and passed it. 

the possible exception we can get is when the user tries to update an existing subscriber through  
 /subscribers/:subscriberid/preferences
 on the GetPreferences usecase flow. 
 wdyt?

Reply:
never mind i can see that you added a check on the [get-subscriber-template-preference.usecase.ts](https://github.com/novuhq/novu/pull/2754/files#diff-cca4106add4716bff4349e085b13e832c6baa61c597d32169309e016e7794cb0). :)